### PR TITLE
Add saving in one transaction feature to Pipe, Joint, ReleaseNote and Spool

### PR DESCRIPTION
**Pull request by serg7**
External files one-transaction in all other places #1313
Add saving in one transaction feature to Pipe, Joint, ReleaseNote and Spool

### CODE REVIEW

**Comment by olga-jane** [comment]
@demmy  please review

File: src/PrizmMainProject/Forms/Component/NewEdit/SaveComponentCommand.cs [hunk 1]
```diff
@@ -89,18 +89,18 @@ public void Execute()
...
                         viewModel.Component.InspectionStatus = viewModel.Component.GetPartInspectionStatus();
                         repos.BeginTransaction();
+                        viewModel.Component.Number = viewModel.Component.Number.ToUpper();
+                        repos.ComponentRepo.SaveOrUpdate(viewModel.Component);
 
                         var filesViewModel = viewModel.FilesFormViewModel;
-
-                        viewModel.Component.Number = viewModel.Component.Number.ToUpper();
-                        repos.ComponentRepo.SaveOrUpdate(viewModel.Component);                      
+                        filesViewModel.DetachFileEntities();
```
Comment (demmy) [inline src/PrizmMainProject/Forms/Component/NewEdit/SaveComponentCommand.cs]: evict before save?

**Comment by demmy** [comment]
Files_NotCopied notify implementet not everywhere

**Comment by serg7** [comment]
Fixed

**Comment by olga-jane** [comment]
@demmy please review again

**Comment by demmy** [comment]
IMHO, OK

**Pull request merged**